### ISSUE TITLE
Long trait impl line is poorly formatted

### ISSUE THREAD

**Issue by Apanatshka** [rust-lang-nursery/rustfmt#1163]
I manually formatted this code as:

```rust
impl<'i, 'a, Input, Payload, A: Automaton<Input, Payload>>
    Iterator for Matches<'i, 'a, Input, Payload, A>
{
    type Item = Match<Payload>;

    fn next(&mut self) -> Option<Self::Item> {
        let mut offset = self.offset;
        while offset < self.input.len() {
            self.state = self.aut.next_state(&self.state, &self.input[offset]);
            offset += 1;
            if let Some(m) = self.aut.get_match(&self.state, 0) {
                self.offset = offset;
                return Some(m);
            }
        }
        None
    }
}
```

`rustfmt --write-mode=diff`:

```diff
-impl<'i, 'a, Input, Payload, A: Automaton<Input, Payload>>⏎
-    Iterator for Matches<'i, 'a, Input, Payload, A>⏎
-{⏎
+impl<'i, 'a, Input, Payload, A: Automaton<Input, Payload>> Iterator for Matches<'i,⏎
+                                                                                'a,⏎
+                                                                                Input,⏎
+                                                                                Payload,⏎
+                                                                                A> {⏎
```

**Comment by sinkuu** [comment]
Try newer version of rustfmt. This should have been fixed by #1148

**Comment by Apanatshka** [comment]
Derp, totally forgot to check for updates :sweat_smile: You're right, it's fixed already.
